holopy/scattering/tests/test_mielensfunctions.py

```diff
@@ -440,6 +441,26 @@ def test_raises_error_on_nans(self):
             parallel_or_perpendicular='perpendicular', **self.default_kwargs)
         self.assertRaises(RuntimeError, interpolator._eval, theta)
 
+    @attr('fast')
+    def test_clips_high_max_l_to_avoid_nans(self):
+        theta = np.array([0.3])
+
+        msm_highl = mielensfunctions.MieScatteringMatrix(
+            parallel_or_perpendicular='perpendicular', max_l=1000,
+            **self.default_kwargs)
+        with warnings.catch_warnings(record=True) as recorded_warnings:
+            warnings.simplefilter('always')
+            s_theta = msm_highl._eval(theta)
+
+        self.assertFalse(np.isnan(s_theta))
+        # Then we check that this test actually worked, and that there
+        # was a nan warning raised:
+        messages = [str(r.message) for r in recorded_warnings]
+        should_be_warned = 'invalid value encountered in cdouble_scalars'
+        present = [should_be_warned in message for message in messages]
+        self.assertTrue(any(present), msg=f'This test is broken: {messages}')
```

Comment:
I feel like there should be a cleaner way to do this, but not sure what it would be.

Reply:
unittest.TestCase.assertWarnsRegex, which I've now used instead. Thanks!

Reply:
(cf 7514e479)